### PR TITLE
Include verbose output for check to list out what needs to be done

### PR DESCRIPTION
**Pull request by technicalpickles**
I mentioned this in https://github.com/Homebrew/homebrew-bundle/issues/176 . I needed a way to figure out why `brew bundle` wasn't installing all the dependencies to satisfy `brew bundle check`.

This can be tested with `brew bundle check --verbose`.

This diff probably could be smaller, but I factored out most of the variables to their own methods so I knew they could be used in the verbose section.

One last thing is: I assume that there's checks like `return false if requested_taps.empty?` to avoid load time / memory for loading the taps tapped if none were requested.

### CODE REVIEW

File: lib/bundle/commands/check.rb [hunk 1]
```diff
@@ -20,36 +39,77 @@ def self.run
...
-    def self.any_formulae_to_install?
-      @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
-      requested_formulae = @dsl.entries.select { |e| e.type == :brew }.map(&:name)
-      requested_formulae.any? do |f|
-        !Bundle::BrewInstaller.formula_installed_and_up_to_date?(f)
+    def self.requested_formulae
+      @requested_formulae ||= dsl.entries.select { |e| e.type == :brew }.map(&:name)
+    end
+
+    def self.formula_to_install
```
Comment (mikemcquaid) [inline lib/bundle/commands/check.rb]: `self.formulae_to_install`

**Comment by mikemcquaid** [comment]
> One last thing is: I assume that there's checks like return false if requested_taps.empty? to avoid load time / memory for loading the taps tapped if none were requested.

Indeed, the goal is to make `brew bundle check` do as little as possible. As soon as it figures out there's missing dependencies it fails. I'm 🆒 with doing more/everything in the verbose case, though.

You've probably seen from CI but just to be explicit: we'll need some more test cases for this.

**Comment by mikemcquaid** [comment]
Oh, and, 👍 and thanks for making a PR 🙇

**Comment by technicalpickles** [comment]
```
  2) Bundle::Commands::Check when formulae are not installed raises an error
     Failure/Error: expect { do_check }.to raise_error(SystemExit)
     
       expected SystemExit, got #<NoMethodError: undefined method `verbose?' for ["--pattern", "spec/**{,/*/**}/*_spec.rb"]:Array> with backtrace:

```

Is there something homebrew/bundle includes that would include `verbose?` on ARGV? I copied this from other commands using it, and it works when run, but not in the tests. Digging more, but figured I'd ask in case someone knows off hand.

**Comment by technicalpickles** [comment]
Looks like it's https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/ARGV.rb . Now to figure out how to get that in testing and tested. It seems there's stubbing that happens like:

```ruby
      allow(ARGV).to receive(:include?).and_return(true)
```

I could just do that for verbose, but that'd end up in several places.

**Comment by MikeMcQuaid** [comment]
> I could just do that for verbose, but that'd end up in several places.

Yeh, that's fine, thanks 👍

**Comment by technicalpickles** [comment]
I haven't gotten a chance to get back to this, and I'm about to go on leave so won't get to it any time soon 😓 

This PR itself isn't super important, but it was useful to help identify what was going on in https://github.com/Homebrew/homebrew-bundle/issues/176

**Comment by MikeMcQuaid** [comment]
Ok, thanks for the heads up. Will pass on this for now.

**Pull request closed without merging**